google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITTracingTest.java

```diff
@@ -0,0 +1,304 @@
+                .build())
+        .buildAndRegisterGlobal();
+  }
+
+  @Before
+  public void before() {
+    FirestoreOptions.Builder optionsBuilder =
+        FirestoreOptions.newBuilder()
+            .setOpenTelemetryOptions(
+                FirestoreOpenTelemetryOptions.newBuilder().setTracingEnabled(true).build());
+    String namedDb = System.getProperty("FIRESTORE_NAMED_DATABASE");
+    if (namedDb != null) {
+      logger.log(Level.INFO, "Integration test using named database " + namedDb);
+      optionsBuilder = optionsBuilder.setDatabaseId(namedDb);
+    } else {
+      logger.log(Level.INFO, "Integration test using default database.");
+    }
+    firestore = optionsBuilder.build().getService();
+
+    // Clean up existing maps.
+    spanNameToSpanId.clear();
+    spanIdToParentSpanId.clear();
+    spanNameToSpanData.clear();
+  }
+
+  @After
+  public void after() throws Exception {
+    Preconditions.checkNotNull(
+        firestore,
+        "Error instantiating Firestore. Check that the service account credentials were properly set.");
+    firestore.shutdown();
+    inMemorySpanExporter.reset();
+  }
+
+  void waitForTracesToComplete() throws Exception {
+    // We need to call `firestore.close()` because that will also close the
+    // gRPC channel and hence force the gRPC instrumentation library to flush
+    // its spans.
+    firestore.close();
+  }
+
+  // Prepares all the spans in memory for inspection.
+  List<SpanData> prepareSpans() throws Exception {
+    waitForTracesToComplete();
+    List<SpanData> spans = inMemorySpanExporter.getFinishedSpanItems();
+    buildSpanMaps(spans);
+    printSpans();
+    return spans;
+  }
+
+  void buildSpanMaps(List<SpanData> spans) {
+    for (SpanData spanData : spans) {
+      spanNameToSpanData.put(spanData.getName(), spanData);
+      spanNameToSpanId.put(spanData.getName(), spanData.getSpanId());
+      spanIdToParentSpanId.put(spanData.getSpanId(), spanData.getParentSpanId());
+    }
+  }
+
+  // Returns the SpanData object for the span with the given name.
+  // Returns null if no span with the given name exists.
+  @Nullable
+  SpanData getSpanByName(String spanName) {
+    return spanNameToSpanData.get(spanName);
+  }
+
+  // Returns the SpanData object for the gRPC span with the given RPC name.
+  // Returns null if no such span exists.
+  @Nullable
+  SpanData getGrpcSpanByName(String rpcName) {
+    return getSpanByName(SERVICE + rpcName);
+  }
+
+  String grpcSpanName(String rpcName) {
+    return SERVICE + rpcName;
+  }
+
+  void assertSameTrace(SpanData... spans) {
+    if (spans.length > 1) {
+      String traceId = spans[0].getTraceId();
+      for (SpanData spanData : spans) {
+        assertEquals(traceId, spanData.getTraceId());
+      }
+    }
+  }
+
+  // Helper to see the spans in standard output while developing tests
+  void printSpans() {
+    for (SpanData spanData : spanNameToSpanData.values()) {
+      System.out.printf(
```

Comment:
maybe switch this to a logging library with levels? That way we don't clutter up the logs when running the ITs

Reply:
Done.